src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -3002,3 +3021,64 @@ def get_aks_custom_headers(aks_custom_headers=None):
                     raise CLIError('custom headers format is incorrect')
                 headers[parts[0]] = parts[1]
     return headers
+
+
+def _get_kubelet_config(file_path):
+    kubelet_config = get_file_json(file_path)
+    if kubelet_config is None:
+        raise CLIError("Error reading kubelet configuration at {}".format(file_path))
+    config_object = KubeletConfig()
+    config_object.cpu_manager_policy = kubelet_config.get("cpuManagerPolicy", None)
+    config_object.cpu_cfs_quota = kubelet_config.get("cpuCfsQuota", None)
+    config_object.cpu_cfs_quota_period = kubelet_config.get("cpuCfsQuotaPeriod", None)
+    config_object.image_gc_high_threshold = kubelet_config.get("imageGcHighThreshold", None)
+    config_object.image_gc_low_threshold = kubelet_config.get("imageGcLowThreshold", None)
+    config_object.topology_manager_policy = kubelet_config.get("topologyManagerPolicy", None)
+    config_object.allowed_unsafe_sysctls = kubelet_config.get("allowedUnsafeSysctls", None)
+    config_object.fail_swap_on = kubelet_config.get("failSwapOn", None)
+
+    return config_object
+
+
+def _get_linux_os_config(file_path):
+    os_config = get_file_json(file_path)
+    if os_config is None:
+        raise CLIError("Error reading Linux OS configuration at {}".format(file_path))
+    config_object = LinuxOSConfig()
+    config_object.transparent_huge_page_enabled = os_config.get("transparentHugePageEnabled", None)
+    config_object.transparent_huge_page_defrag = os_config.get("transparentHugePageDefrag", None)
+    config_object.swap_file_size_mb = os_config.get("swapFileSizeMB", None)
+    # sysctl settings
+    sysctls = os_config.get("sysctls", None)
+    if sysctls is not None:
```

Comment:
How about use `if sysctls:` to cover other empty case (empty string, dict, array, etc) ?

Reply:
updated to use isinstance, this actually has to be a dict otherwise get would throw.